hiro.proto

```diff
@@ -2257,3 +2266,87 @@ message SyncResponse {
   // Current server time.
   int64 current_time_sec = 11;
 }
+
+// Represents an available reward based on the progress of a streak.
+message StreakAvailableReward {
+  // Minimum streak progress count required for this reward, inclusive.
+  int64 count_min = 1;
+  // Maximum streak progress count required for this reward, inclusive.
+  int64 count_max = 2;
+  // Available reward contents.
+  AvailableRewards reward = 3;
+}
+
+// A single claimed streak reward.
+message StreakReward {
+  // Minimum streak progress count required for this reward, inclusive.
+  int64 count_min = 1;
+  // Maximum streak progress count required for this reward, inclusive.
+  int64 count_max = 2;
+  // Reward that was actually granted.
+  Reward reward = 3;
+  // The UNIX time (for gRPC clients) or ISO string (for REST clients) when this reward was claimed.
+  int64 claim_time_sec = 4;
+}
+
+// An individual streak, along with its status and progress if any.
+message Streak {
+  // Unique identifier.
+  string id = 1;
+  // Display name for this streak.
+  string name = 2;
+  // Current overall progress count.
+  int64 count = 3;
+  // Maximum allowed overall progress count.
+  int64 max_count = 4;
+  // Progress count submitted during this reset.
+  int64 count_current_reset = 5;
+  // Maximum progress count allowed during any single reset.
+  int64 max_count_current_reset = 6;
+  // How much overall count is lost per idle reset.
+  int64 idle_count_decay_reset = 7;
```

Comment:
Is the reset for when a user has passed the `reset_time_sec` without a login? So is `count_current_reset` the last streak they maintained before a reset? What is the `max_count_current_reset` for? and is the `idle_count_decay_reset` the value to remove on reset? If left empty does it do a full reset to zero?

Reply:
A reset decay happens when a full reset passes without the user having submitted any updates to this streak. The `count_current_reset` is the count submitted during the current reset only. `max_count_current_reset` is the maximum count that a user can submit during a single reset. `idle_count_decay_reset` is the value that is removed per missed reset (so missing 2 resets in a row would remove `2 * idle_count_decay_reset`) and if left empty indicates no decay.

Reply:
cool, thanks for the clarification